internal/core/metadata/rest_device.go

```diff
@@ -569,39 +523,29 @@ func restSetDeviceAdminStateById(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Update the AdminState
-	d.AdminState = newAs
-	if err = dbClient.UpdateDevice(d); err != nil {
+	if err = updateDeviceState(updateMode, state, d); err != nil {
 		LoggingClient.Error(err.Error())
-		http.Error(w, err.Error(), http.StatusServiceUnavailable)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	ctx := r.Context()
-	if err := notifyDeviceAssociates(d, http.MethodPut, ctx); err != nil {
-		LoggingClient.Error(err.Error())
-		http.Error(w, err.Error(), http.StatusServiceUnavailable)
-		return
-	}
+	// Notify
+	notifyDeviceAssociates(d, http.MethodPut, r.Context())
```

Comment:
I'm not %100 on board with this change, but I'm not going to block the PR because of it. I'm assuming you're ignoring the error because this function calls other services, and those subsequent actions don't really have a bearing on the success/failure of updating the device's state. I also think the internals of `notifyDeviceAssociates` needs significant rework -- utilizing concurrency for posting the notification and then invoking the callbacks through `notifyAssociates`. That can all be done as a separate issue.